processor/src/decoder/tests.rs

```diff
@@ -877,6 +877,202 @@ fn loop_block_repeat() {
+    check_op_decoding(&dec_trace, 4, span1_addr, Operation::FmpUpdate, 0, 1, 1);
+    check_op_decoding(&dec_trace, 5, span1_addr, Operation::End, 0, 0, 0);
+    // starting CALL block
+    let call_addr = span1_addr + EIGHT;
+    check_op_decoding(&dec_trace, 6, join1_addr, Operation::Call, 0, 0, 0);
+    // starting second SPAN block
+    let span2_addr = call_addr + EIGHT;
+    check_op_decoding(&dec_trace, 7, call_addr, Operation::Span, 2, 0, 0);
+    check_op_decoding(&dec_trace, 8, span2_addr, Operation::Push(ONE), 1, 0, 1);
+    check_op_decoding(&dec_trace, 9, span2_addr, Operation::FmpUpdate, 0, 1, 1);
+    check_op_decoding(&dec_trace, 10, span2_addr, Operation::End, 0, 0, 0);
+    // ending CALL block
+    check_op_decoding(&dec_trace, 11, call_addr, Operation::End, 0, 0, 0);
+    // ending internal JOIN block
+    check_op_decoding(&dec_trace, 12, join1_addr, Operation::End, 0, 0, 0);
+    // starting the 3rd SPAN block
+    let span3_addr = span2_addr + EIGHT;
+    check_op_decoding(&dec_trace, 13, INIT_ADDR, Operation::Span, 1, 0, 0);
+    check_op_decoding(&dec_trace, 14, span3_addr, Operation::FmpAdd, 0, 0, 1);
+    check_op_decoding(&dec_trace, 15, span3_addr, Operation::End, 0, 0, 0);
+    // ending the program
+    check_op_decoding(&dec_trace, 16, INIT_ADDR, Operation::End, 0, 0, 0);
+    check_op_decoding(&dec_trace, 17, ZERO, Operation::Halt, 0, 0, 0);
+
+    // --- check hasher state columns -------------------------------------------------------------
+    // in the first row, the hasher state is set to hashes of (join1, span3)
+    let join1_hash: Word = join1.hash().into();
+    let span3_hash: Word = span3.hash().into();
+    assert_eq!(join1_hash, get_hasher_state1(&dec_trace, 0));
+    assert_eq!(span3_hash, get_hasher_state2(&dec_trace, 0));
+
+    // in the second row, the hasher state is set to hashes of (span1, fn_block)
+    let span1_hash: Word = span1.hash().into();
+    let fn_block_hash: Word = fn_block.hash().into();
+    assert_eq!(span1_hash, get_hasher_state1(&dec_trace, 1));
+    assert_eq!(fn_block_hash, get_hasher_state2(&dec_trace, 1));
+
+    // at the end of the first SPAN, the hasher state is set to the hash of the first child
+    assert_eq!(span1_hash, get_hasher_state1(&dec_trace, 5));
+    assert_eq!([ZERO, ZERO, ZERO, ZERO], get_hasher_state2(&dec_trace, 5));
+
+    // in the 7th row, we start the CALL block which hash span2 as its only child
+    let span2_hash: Word = span2.hash().into();
+    assert_eq!(span2_hash, get_hasher_state1(&dec_trace, 6));
+    assert_eq!([ZERO, ZERO, ZERO, ZERO], get_hasher_state2(&dec_trace, 5));
+
+    // span2 ends in the 11th row
+    assert_eq!(span2_hash, get_hasher_state1(&dec_trace, 10));
+    assert_eq!([ZERO, ZERO, ZERO, ZERO], get_hasher_state2(&dec_trace, 10));
+
+    // CALL block ends in the 12th row; the second to last element of the hasher state
+    // is set to ONE because we are existing the CALL block
```

Comment:
nit: I think there's a typo here. Should it be "exiting" instead of "existing"?

Reply:
Fixed!